spring-core/src/main/java/org/springframework/util/comparator/BooleanComparator.java

```diff
@@ -76,7 +78,7 @@ public boolean equals(@Nullable Object other) {
 
 	@Override
 	public int hashCode() {
-		return getClass().hashCode() * (this.trueLow ? -1 : 1);
```

Comment:
since `java-8` there is a way to get the hashCode without calling into `getClass`